src/hammer-vlsi/defaults.yml

```diff
@@ -449,6 +449,29 @@ vlsi.inputs:
     #   Requires the semi_auto pin generation mode.
     assignments: []
 
+  # SVG visualization of placement constraints and bumps for the current hierarchical module.
+  # Used by the generate_visualization method accessible by all HammerTools.
+  visualization:
+    # Type of visualization to draw
+    # Note: pcb action will mirror visualization across y-axis (assumes flip-chip on PCB)
+    # Valid options:
+    # - all: draws all placement constraints, bumps, and pad designators (if running pcb action)
+    # - floorplan: draws only placement constraints
+    # - bumps: draws only bumps
+    # - footprint: draws PCB footprint, only valid if running pcb action
+    # type: str
+    mode: "all"
+    # Name of the desired output SVG file
+    # type: str
+    svg_file: "${vlsi.inputs.visualization.mode}.svg"
+    svg_file_meta: lazysubst
+    # Depth of paths to shorten for better readability in the SVG
+    # E.g. top/path/to/inst will shorten to t/p/t/inst for depth of 1 and t/p/top/inst for depth of 2
+    # A depth of 0 or less will perform no path shortening.
+    # type: int
+    shorten_path_depth: 1
```

Comment:
Is it really useful to have the `t/p/t` portions of the path?
I guess I'm imagining the hierarchical paths getting very long in submodules and maybe it would be better to have just `top/path/to/submodule/path/to/inst` become `s/p/t/inst`

Reply:
It already is `s/p/t/inst` for submodules because the first letter of the path will always be the current hierarchical module. So using your example:

Flat flow: `t/p/t/s/p/t/inst`
Hier flow, viz in `submodule`: `s/p/t/inst`
Hier flow, viz in top module: `t/p/t/submodule_inst`

Reply:
Ok I guess we should just think about this again when/if we do the nested viz